src/neptune/metadata_containers/run.py

```diff
@@ -451,6 +461,43 @@ def _write_initial_attributes(self):
             # upload default sources ONLY if creating a new run
             upload_source_code(source_files=self._source_files, run=self)
 
+        if self._dependencies:
+            self._track_dependencies()
+
+    def _track_dependencies(self) -> None:
+        if self._dependencies == "infer":
+            from neptune.internal.utils.logger import logger
+
+            logger.info("INFO: Calling pipreqs to generate requirements based on project imports.")
+            # using pipreqs here
+            try:
+                proc = subprocess.Popen(["pipreqs", "--print", "."], stdout=subprocess.PIPE)
```

Comment:
Check this: https://stackoverflow.com/a/4760517
This current solution is probably a bit too complicated and low-level